build/android/app/build.gradle

```diff
@@ -0,0 +1,106 @@
+apply plugin: 'com.android.application'
+android {
+	compileSdkVersion 29
+	buildToolsVersion '29.0.2'
+	defaultConfig {
+		applicationId 'net.minetest.minetest'
+		minSdkVersion 16
+		targetSdkVersion 29
+		versionName "${versionMajor}.${versionMinor}.${versionPatch}"
+		versionCode project.versionCode
+	}
+
+	Properties props = new Properties()
+	props.load(new FileInputStream(file('../local.properties')))
+
+	if (props.getProperty('keystore') != null) {
+		signingConfigs {
+			release {
+				storeFile file(props['keystore'])
+				storePassword props['keystore.password']
+				keyAlias props['key']
+				keyPassword props['key.password']
+			}
+		}
+
+		buildTypes {
+			release {
+				minifyEnabled true
+				signingConfig signingConfigs.release
+			}
+		}
+	}
+
+	// for multiple APKs
+	splits {
+		abi {
+			enable true
+			reset()
+			include 'armeabi-v7a', 'arm64-v8a'
+		}
+	}
+
+	compileOptions {
+		sourceCompatibility JavaVersion.VERSION_1_8
+		targetCompatibility JavaVersion.VERSION_1_8
+	}
+}
+
+task prepareAssets() {
```

Comment:
Is CMake not used at all to build for Android?

Reply:
Not now. Now we will get rid of the obsolete Makefile, and add build support for all OS. CMake will be added in the future.